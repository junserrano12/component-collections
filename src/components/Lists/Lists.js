import React from 'react'
import PropTypes from 'prop-types'
import style from './Lists.scss'

const renderChildren = (children) => {
    return React.Children.map(children, child => {
        if (child.type === List)
            return child
        else
            return null
    })
}

const List = ({classes=[], children}) => {

    let contentStyle = `${style.content}`

    classes.map( (classNames) => {
        contentStyle += ` ${classNames}`
    })

    return (
        <li className={contentStyle}>
            {children}
        </li>
    )
}

const Lists = ({children, type='default', classes=[]}) => {
    
    let containerStyle = `${style.container} ${type}`

    classes.map(classNames => {
        containerStyle += ` ${classNames}`
    })

    return (
        <ul className={containerStyle}>
            {renderChildren(children)}
        </ul>
    )
}

Lists.propTypes = {
    type: PropTypes.string,
    classes: PropTypes.array
}

List.propTypes = {
    keyName: PropTypes.string,
    classes: PropTypes.array
}

export default Lists

export {List}