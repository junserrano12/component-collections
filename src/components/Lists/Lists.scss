.container {
    display: block;
    position: relative;
    margin: 0;
    padding: 0;    

    &[class*="bullet"] {
        list-style: disc outside;
        margin-left: 24px;
    }    
    
    &[class*="default"] {
        list-style: none;
    }
    
}

.content {
    position: relative;
}