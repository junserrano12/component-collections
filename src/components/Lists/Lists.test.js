import React from 'react'
import Lists, {List} from './Lists'
import Enzyme, {shallow} from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

Enzyme.configure({adapter: new Adapter()})

describe('Lists component', () => {

    it('Should render ul tag and its children', () => {
        const component = shallow(<Lists><List key="001">Content</List></Lists>)
        expect(component).toMatchSnapshot()
    })

    it('Should not render other components except for List component', () => {
        const children = [
            <List key="001">content 1</List>,
            <List key="002">content 2</List>,
            <p key="003">other component</p>
        ]
        const component = shallow(<Lists>{children}</Lists>)
        expect(component).toMatchSnapshot()
    })

    it('Should have a class="container bullet" if props.type is set "bullet"', () => {
        const children = [
            <List key="001">content 1</List>
        ]
        const component = shallow(<Lists type="bullet">{children}</Lists>)
        expect(component).toMatchSnapshot()
    })

    it('Should have class="container default foo bar" if props.classes is set ["foo", "bar"]', () => {
        const children = [
            <List key="001">content 1</List>
        ]
        const component = shallow(<Lists classes={["foo", "bar"]}>{children}</Lists>)
        expect(component).toMatchSnapshot()
    })

})

describe('List Component', () => {

    it('Should render li tag and its children', () => {
        const component = shallow(<List>Content</List>)
        expect(component).toMatchSnapshot()
    }) 

    it('Should have class="content foo bar" if props.classes is set ["foo", "bar"]', () => {
        const component = shallow(<List classes={["foo", "bar"]}>Content</List>)
        expect(component).toMatchSnapshot()
    })

})