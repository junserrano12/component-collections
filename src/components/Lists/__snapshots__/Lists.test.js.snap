// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Component Should have class="content foo bar" if props.classes is set ["foo", "bar"] 1`] = `
<li
  className="undefined foo bar"
>
  Content
</li>
`;

exports[`List Component Should render li tag and its children 1`] = `
<li
  className="undefined"
>
  Content
</li>
`;

exports[`Lists component Should have a class="container bullet" if props.type is set "bullet" 1`] = `
<ul
  className="undefined bullet"
>
  <List
    key=".$001"
  >
    content 1
  </List>
</ul>
`;

exports[`Lists component Should have class="container default foo bar" if props.classes is set ["foo", "bar"] 1`] = `
<ul
  className="undefined default foo bar"
>
  <List
    key=".$001"
  >
    content 1
  </List>
</ul>
`;

exports[`Lists component Should not render other components except for List component 1`] = `
<ul
  className="undefined default"
>
  <List
    key=".$001"
  >
    content 1
  </List>
  <List
    key=".$002"
  >
    content 2
  </List>
</ul>
`;

exports[`Lists component Should render ul tag and its children 1`] = `
<ul
  className="undefined default"
>
  <List
    key=".$001"
  >
    Content
  </List>
</ul>
`;
